### DISCUSSION TITLE
Switch language, Custom Modal dialog, Passing Parameters, RTL

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#2131] [Q&A]
Hi,

I'm really satisfied with Oqtane Module development.

Trying to implement "Switch language, Custom Modal dialog, Passing Parameters between Modules & RTL layout" in my project Master module.

1) How do I implement language switch which is in Theme Header Component and get currently selected language in my modules?

2) How do I do it - If user clicks edit button in Radzen Datagrid component and it shows editable data in Radzen Modal dialog? **Note:** Currently Oqtane's ActionDialog component hiding Datagrid while on edit.

3) How do I do it - passing parameters or data between Modules. Example, if user change dropdownlist selected value in module1 then (there should be API call with parameters to the backend and it's resultant) based on selected value some data binding to Datagrid in module2?

4) How do I do it - Switch between LTR to RTL and vice versa if current language culture changes?

**Comment by hishamco** [comment]
There's already language switch if you enable more than one language. For switching the layout  to RTL or LTR I think this is not supported out of the box but it's doable

**Reply by Mostafa-Moafi** [reply]
Does this CMS support RTL?
If it doesn't support can you teach me how to RTL it?

**Reply by hishamco** [reply]
Yep, RTL is supported

**Reply by hishamco** [reply]
Languages Management you can add Arabic for instance, then you can use language switcher to change the language

**Reply by leigh-pointer** [reply]
@Mostafa-Moafi indeed it is, when you create a rtl language pack it will render correctly! Personally it would be nice to see, so good luck and support is here if you need it!

**Reply by Mostafa-Moafi** [reply]
Thanks @hishamco , @leigh-pointer

**Comment by sbwalker** [comment]
For # 3 please read - https://www.oqtane.org/blog/!/40/inotifypropertychanged as it describes how to pass data between modules (this capability is used in a number of areas in the core framework). There is also a Demo31 repo which demonstrates the feature.

**Comment by sbwalker** [comment]
For # 2 you will need to provide a screen shot as it difficult to understand what you trying to do
